src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/ResourceMonitoringServiceCollectionExtensions.cs

```diff
@@ -62,7 +62,7 @@ public static class ResourceMonitoringServiceCollectionExtensions
 #if NETFRAMEWORK
         _ = builder.AddWindowsProvider();
 #else
-        if (GetPlatform() == PlatformID.Win32NT)
+        if (OperatingSystem.IsWindows())
```

Comment:
In that case we need to check whether `else` clause falls under `Linux`  - and throw an exception otherwise

Reply:
Added.